src/buddy_fetch.py

```diff
@@ -64,37 +65,37 @@ def get_shortened_name(full_hero_name):
 
 
 def get_buddy_dict():
-    # dictionary to hold hero names and their ID's
-    battlegrounds_heroes = {}
+    # dictionary to hold hero ID's and their names
+    _heroes = {}
     # Variable of the final dictionary
-    battlegrounds_heroes_buddies = {}
-
-    # Read through all the data and find the hero names and their ID's
-    for i in range(len(data_json)):
-        if "battlegroundsHero" in data_json[i].keys():
-            battlegrounds_heroes[data_json[i]["name"]] = data_json[i]["id"]
-
-    # Loop through all the heroes in the dictionary and find their buddies
-    for key in battlegrounds_heroes:
-        for j in range(len(data_json)):
-            if "isBattlegroundsBuddy" in data_json[j].keys():
-                if data_json[j]["id"] == battlegrounds_heroes[key] + "_Buddy":
-                    data = data_json[j]
-                    buddy_string = f"{data['name']} is a Tier {data['techLevel']} {data['attack']}/{data['health']}. Ability: {filterText(data['text'])}"
-                    battlegrounds_heroes_buddies[get_shortened_name(key)] = (
-                        key,
-                        buddy_string,
-                    )
-                elif data_json[j]["id"] == battlegrounds_heroes[key] + "_Buddy_G":
-                    data = data_json[j]
-                    golden_buddy_string = f"Golden {data['name']} is a Tier {data['techLevel']} {data['attack']}/{data['health']}. Ability: {filterText(data['text'])}"
-                    battlegrounds_heroes_buddies[
-                        get_shortened_name(key)
-                    ] = battlegrounds_heroes_buddies[get_shortened_name(key)] + (
-                        golden_buddy_string,
-                    )
-
-    return battlegrounds_heroes_buddies
+    buddies = {}
+
+    # Loop through the heroes in data_json, and find their ID's and names
+    for hero in filter(lambda card: "battlegroundsBuddyDbfId" in card, data_json):
+        id_words = hero["id"].split("_")
+        # verify that the card is not a skin
+        if len(id_words) >= 2 and id_words[-2] == "HERO" and id_words[-1].isnumeric():
+            _heroes[hero["id"]] = hero["name"]
+
+    # Loop through the buddies in data_json and find their heroes
+    for buddy in filter(lambda card: "isBattlegroundsBuddy" in card, data_json):
```

Comment:
legacy code was looping through data.json for every hero, now it only loops once through data.json